annotation/tests/test_tasks_services.py

```diff
@@ -295,3 +303,168 @@ def test_validate_user_actions_invalid_states(
+
+    mock_query.filter_by.assert_called_once()
+    mock_query.limit.assert_called_once_with(10)
+
+
+@pytest.mark.parametrize(
+    ("search_id", "search_name", "ids_with_names", "expected_result"),
+    (
+        (1, "Task 1", {1: "Task 1", 2: "Task 2"}, ([1], {1: "Task 1"})),
+        (1, None, {1: "Task 1", 2: "Task 2"}, ([1], {})),
+    ),
+)
+def test_validate_ids_and_names(
+    search_id: int,
+    search_name: str,
+    ids_with_names: Dict[int, str],
+    expected_result: Tuple[Optional[List[int]], Dict[int, str]],
+):
+    result = validate_ids_and_names(
+        search_id=search_id,
+        search_name=search_name,
+        ids_with_names=ids_with_names,
+    )
+    assert result == expected_result
+
+
+@pytest.mark.parametrize(
+    ("search_id", "search_name", "ids_with_names"),
+    (
+        (3, "Task 3", {1: "Task 1", 2: "Task 2"}),
+        (1, "Task 1", {}),
+        (None, "Task 2", None),
+    ),
+)
+def test_validate_ids_and_names_invalid_name_or_id(
+    search_id: int, search_name: str, ids_with_names: Dict[int, str]
+):
+    with pytest.raises(HTTPException) as exc_info:
+        validate_ids_and_names(
+            search_id=search_id,
+            search_name=search_name,
+            ids_with_names=ids_with_names,
+        )
+    assert exc_info.value.status_code == 404
+
+
+def test_remove_additional_filters_with_standard_filters():
+    filter_args = {
+        "filters": [
+            {"field": "normal_field", "value": "value1"},
+            {"field": "normal_field_2", "value": "value2"},
+        ],
+        "sorting": [{"field": "normal_field", "order": "asc"}],
+    }
+    expected_filters = {
+        "filters": [
+            {"field": "normal_field", "value": "value1"},
+            {"field": "normal_field_2", "value": "value2"},
+        ],
+        "sorting": [{"field": "normal_field", "order": "asc"}],
+    }
+    expected_additional_filters = {}
+
+    result = remove_additional_filters(filter_args)
+
+    assert filter_args == expected_filters
+    assert result == expected_additional_filters
+
+
+def test_remove_additional_filters_with_additional_fields():
+    filter_args = {
+        "filters": [
+            {"field": "file_name", "value": ["file1", "file2"]},
+            {"field": "job_name", "value": "job1"},
+        ],
+        "sorting": [],
+    }
+    expected_filters = {
+        "filters": [],
+        "sorting": [],
+    }
+    expected_additional_filters = {
+        "file_name": ["file1", "file2"],
+        "job_name": ["job1"],
+    }
+
+    result = remove_additional_filters(filter_args)
+
+    assert filter_args == expected_filters
+    assert result == expected_additional_filters
+
+
+@pytest.mark.parametrize(
+    ("task_id", "tenant", "expected_result"),
+    ((1, "tenant_1", "task_1"), (2, "tenant_2", None)),
+)
+def test_read_annotation_task(
+    mock_session: Mock,
+    task_id: int,
+    tenant: str,
+    expected_result: Union[str, None],
```

Comment:
As I said before, Union[something, None] == Optional[something]. Utilize it

Reply:
Ok.